crates/router/src/connector/bluesnap.rs

```diff
@@ -0,0 +1,741 @@
+        Ok(msg.into_bytes())
+    }
+
+    async fn get_webhook_source_verification_merchant_secret(
+        &self,
+        db: &dyn StorageInterface,
+        merchant_id: &str,
+    ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
+        let key = format!("whsec_verification_{}_{}", self.id(), merchant_id);
+        let secret = db
+            .find_config_by_key(&key)
+            .await
+            .change_context(errors::ConnectorError::WebhookVerificationSecretNotFound)?;
+
+        Ok(secret.config.into_bytes())
+    }
+
+    async fn verify_webhook_source(
+        &self,
+        db: &dyn StorageInterface,
+        headers: &actix_web::http::header::HeaderMap,
+        body: &[u8],
+        merchant_id: &str,
+    ) -> CustomResult<bool, errors::ConnectorError> {
+        let algorithm = self
+            .get_webhook_source_verification_algorithm(headers, body)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+
+        let signature = self
+            .get_webhook_source_verification_signature(headers, body)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        let mut secret = self
+            .get_webhook_source_verification_merchant_secret(db, merchant_id)
+            .await
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        let mut message = self
+            .get_webhook_source_verification_message(headers, body, merchant_id, &secret)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        message.append(&mut secret);
+        algorithm
+            .verify_signature(&secret, &signature, &message)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)
+    }
+
+    fn get_webhook_object_reference_id(
+        &self,
+        body: &[u8],
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let webhook_body: bluesnap::BluesnapWebhookBody = serde_urlencoded::from_bytes(body)
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookSignatureNotFound)?;
+        Ok(webhook_body.reference_number)
+    }
+
+    fn get_webhook_event_type(
+        &self,
+        body: &[u8],
+    ) -> CustomResult<api::IncomingWebhookEvent, errors::ConnectorError> {
+        let details: bluesnap::BluesnapWebhookObjectEventType = serde_urlencoded::from_bytes(body)
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
+
+        Ok(match details.transaction_type.as_str() {
+            "DECLINE" | "CC_CHARGE_FAILED" => api::IncomingWebhookEvent::PaymentIntentFailure,
+            "CHARGE" => api::IncomingWebhookEvent::PaymentIntentSuccess,
+            _ => Err(errors::ConnectorError::WebhookEventTypeNotFound).into_report()?,
+        })
+    }
+
+    fn get_webhook_resource_object(
+        &self,
+        body: &[u8],
```

Comment:
shouldn't this be of type `&api::IncomingWebhookRequestDetails<'_>` instead of `&[u8]`, are you not getting compilation error?